crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_type_union.rs

```diff
@@ -74,15 +92,83 @@ pub(crate) fn unnecessary_type_union<'a>(checker: &mut Checker, union: &'a Expr)
-                    .collect(),
+                members: type_members.clone(),
                 is_pep604_union,
             },
             union.range(),
-        ));
+        );
+
+        if checker.patch(diagnostic.kind.rule()) {
+            let union_str = if is_pep604_union {
+                checker
+                    .generator()
+                    .expr(&Expr::Subscript(ast::ExprSubscript {
+                        value: Box::new(Expr::Name(ast::ExprName {
+                            id: "type".into(),
+                            ctx: ExprContext::Load,
+                            range: TextRange::default(),
+                        })),
+                        slice: Box::new(concatenate_bin_ors(
+                            type_exprs
+                                .clone()
+                                .into_iter()
+                                .map(std::convert::AsRef::as_ref)
+                                .collect(),
+                        )),
+                        ctx: ExprContext::Load,
+                        range: TextRange::default(),
+                    }))
+            } else {
+                checker
+                    .generator()
+                    .expr(&Expr::Subscript(ast::ExprSubscript {
+                        value: Box::new(Expr::Name(ast::ExprName {
+                            id: "type".into(),
+                            ctx: ExprContext::Load,
+                            range: TextRange::default(),
+                        })),
+                        slice: Box::new(Expr::Subscript(ast::ExprSubscript {
+                            value: Box::new(Expr::Name(ast::ExprName {
+                                id: "Union".into(),
+                                ctx: ExprContext::Load,
+                                range: TextRange::default(),
+                            })),
+                            slice: Box::new(Expr::Tuple(ast::ExprTuple {
+                                elts: type_members
+                                    .into_iter()
+                                    .map(|type_member| {
+                                        Expr::Name(ast::ExprName {
+                                            id: type_member,
+                                            ctx: ExprContext::Load,
+                                            range: TextRange::default(),
+                                        })
+                                    })
+                                    .collect(),
+                                ctx: ExprContext::Load,
+                                range: TextRange::default(),
+                            })),
+                            ctx: ExprContext::Load,
+                            range: TextRange::default(),
+                        })),
+                        ctx: ExprContext::Load,
+                        range: TextRange::default(),
+                    }))
```

Comment:
is there a shorthand for any of this?

Reply:
No, it's bad 😞

Reply:
That is _NOT_ what I wanted to hear! 🤣 Thanks anyways.

Reply:
Hopefully clear that I mean the abstractions are bad, and not your code :)

Reply:
🤣 of course.

...but why not both?